20231018_signal_cmb/maps.py

```diff
@@ -48,11 +51,15 @@ def _build_maps(idx, beam_amin: float, nside:int, job='TQU'):
     maps = np.empty((ncomp, hp.nside2npix(nside)), dtype=float)
     if 'T' in job.upper():
         hp.almxfl(teb[0], bl_T * pw_T, inplace=True)
-        maps[0] = hp.alm2map(teb[0], nside)
+        maps_ = hp.alm2map(teb[0], nside)
+        maps[0] = rot.rotate_map_pixel(maps_) if coord else maps_
+        del maps_
     if 'QU' in job.upper():
         hp.almxfl(teb[1], bl_E * pw_P, inplace=True)
         hp.almxfl(teb[2], bl_B * pw_P, inplace=True)
-        maps['T' in job.upper():] = hp.alm2map_spin(teb[1:], nside, 2, lmax)
+        maps_ = hp.alm2map(teb, nside)
+        maps['T' in job.upper():] = rot.rotate_map_pixel(maps_)[1:] if coord else maps_[1:]
```

Comment:
any reason you are using pixel space rotation?
we are already in Alm space, we could do a `rotate_alm`

Reply:
I will change it to alms, I just realised that, rotate_alm has the option for lmax